File: src/main/java/phoenix/client/gui/diaryPages/elements/RightAlignedTextElement.java
package phoenix.client.gui.diaryPages.elements;

import net.minecraft.client.gui.FontRenderer;
import net.minecraft.client.gui.screen.inventory.ContainerScreen;
import phoenix.containers.DiaryContainer;

public class RightAlignedTextElement extends TextElement
{
    public RightAlignedTextElement(String text)
    {
        super(text);
    }

    public RightAlignedTextElement(String text, Integer color)
    {
        super(text, color);
    }

    @Override
    public void render(ContainerScreen<DiaryContainer> gui, FontRenderer font, int xSize, int ySize, int x, int y, int depth)
    {
        super.render(gui, font, xSize, ySize, xSize - font.getStringWidth(text.getFormattedText()), y, depth);
    }

    @Override
    public String toString()
    {
        return "[r]" + text + "[R]";  
    }
}

File: src/main/kotlin/phoenix/client/gui/diary/elements/RightAlignedTextElement.kt
package phoenix.client.gui.diary.elements

import net.minecraft.client.gui.FontRenderer
import phoenix.client.gui.DiaryGui

class RightAlignedTextElement : TextElement
{
    constructor(text: String) : super(text)
    {
    }

    constructor(text: String, color: Int) : super(text, color)
    {
    }

    override fun render(
        gui: DiaryGui,
        font: FontRenderer,
        xSize: Int,
        ySize: Int,
        x: Int,
        y: Int,
        depth: Int
    )
    {
        super.render(gui, font, xSize, ySize, x + xSize - 25 - font.getStringWidth(text.formattedText), y, depth)
    }

    override fun toString(): String
    {
        return "[r]$text[R]"
    }
}